app/helpers/ems_physical_infra_helper/textual_summary.rb

```diff
@@ -62,9 +62,13 @@ def textual_physical_servers
     h
   end
 
-  def textual_hosts
+  def textual_physical_servers_with_host
     count_of_host_relationships = (@record.physical_servers.select { |server| !server.host.nil? }).length
-    {:label => _("Hosts"), :icon => "pficon pficon-screen", :value => count_of_host_relationships}
+    h = {:label => _("Physical Servers with Host"), :icon => "pficon pficon-server", :value => count_of_host_relationships}
+    if count_of_host_relationships > 0
+      h[:link] = "/ems_physical_infra/#{@ems.id}?display=physical_servers&options=physical_servers_with_host"
```

Comment:
I don't see the place where the model lookup is done, can you point me there, please?

I mean we can have:

```
 def display_physical_servers
     nested_list('physical_servers', PhysicalServer, :breadcrumb_title => _("Physical Servers"))
 end

 def  display_physical_servers_with_host
      nested_list('physical_servers', PhysicalServer,
                  :where_clause     => "physical_servers.id in (select hosts.physical_server_id from hosts)",
                  :breadcrumb_title => _("Physical Servers with Host"))
 end
```
and then
```
  def self.display_methods                                                                          
    %w(physical_servers physical_servers_with_host)                                                                    
  end                                                                                               
```       

can't we?

And looking at `(@record.physical_servers.select { |server| !server.host.nil? }).length` I wonder how that evaluates and what SQL queries are defined. Can that be expressed as some (cheeper) AREL query that could result in an SQL `count(...)` query?